packages/ingest/src/queue.js

```diff
@@ -58,6 +59,7 @@ async function enqueueParsePdrMessage({
     customMeta: {
       collection,
       provider,
+      ...additionalCustomMeta,
```

Comment:
Is it dangerous that this would in theory let people override `meta.collection` and `meta.provider`?

Reply:
It might be if we want to be sure that people can't mess up `meta.collection` and `meta.provider`. The same goes for the other function with this logic

Reply:
It may even be worth a unit test to ensure that people can't mess that up

Reply:
https://github.com/nasa/cumulus/pull/2054/commits/a8d373ba3c39f00b95c21004633a0328739dfb0f reordered logic and added unit tests

Reply:
Confirmed in ad-hoc testing that the new units break if the original ordering is used.

Reply:
This doesn't feel great, largely because we're now, as part of the Core message code, specifying reserved/non-reserved fields in .meta for modification by <things> and not *it won't work right if you do this*, but via code constraints.   It's not clear that squares with: https://nasa.github.io/cumulus/docs/workflows/cumulus-task-message-flow#docsNav

I think we need to redefine ^ and/or consider *namespacing* the values in meta, or perhaps explicitly move provider/collection into cumulus_meta. 

I'm not sure that's a legit concern in merging this PR, but we should probably have a ticket on the workplan or at *least* an arch topic to discuss this.

Reply:
(and apologies for the discussion necromancy)